### PR TITLE
[tests] Remove 'account' API from wallet functional tests

### PR DESCRIPTION
**Pull request by jnewbery**
Next step in #12952. Removes all usage of the 'account' API from the wallet functional tests, except:

- rpc_deprecated.py (which specifically tests the `-deprecatedrpc=accounts` command line argument is working properly).
- `wallet_labels.py` (which tests that both the 'label' and 'account' APIs work in V0.17).

'account' API usage for both of those tests can be removed once V0.17 has been branched.

Also excluded is:

- `wallet_importprunedfunds.py` (which fails due to a bitcoind OOM error)

### CODE REVIEW

**Comment by laanwj** [comment]
Nice!
Concept ACK.

**Comment by jonasschnelli** [comment]
Concept ACK.
I think combining some commits (especially the flake8 fixes) would help in review.

**Comment by laanwj** [comment]
> This currently fails Travis due to an OOM issue exposed by wallet_importprunedfunds.py

Ref: this issue is tracked in #13078, and so this PR depends on that being fixed

**Comment by jnewbery** [comment]
I've squashed all the flake8 fixes into a single commit as requested by @jonasschnelli , and opened a PR for just those fixes at #13136.

File: test/functional/wallet_txn_doublespend.py [hunk 1]
```diff
@@ -36,20 +35,20 @@ def run_test(self):
...
-        node0_address_bar = self.nodes[0].getnewaddress("bar")
-        fund_bar_txid = self.nodes[0].sendfrom("", node0_address_bar, 29)
+        node0_address_bar = self.nodes[0].getnewaddress()
+        fund_bar_txid = self.nodes[0].sendtoaddress(node0_address_bar, 29)
         fund_bar_tx = self.nodes[0].gettransaction(fund_bar_txid)
 
-        assert_equal(self.nodes[0].getbalance(""),
-                     starting_balance - 1219 - 29 + fund_foo_tx["fee"] + fund_bar_tx["fee"])
+        assert_equal(self.nodes[0].getbalance(),
+                     starting_balance + fund_foo_tx["fee"] + fund_bar_tx["fee"])
```
Comment (MarcoFalke) [inline test/functional/wallet_txn_doublespend.py]: Why this change?

**Comment by MarcoFalke** [comment]
tests fail on travis

**Comment by jnewbery** [comment]
Thanks @MarcoFalke . Should now be fixed.

**Comment by sipa** [comment]
Concept ACK

File: test/functional/wallet_keypool_topup.py [hunk 2]
```diff
@@ -25,7 +25,7 @@ class KeypoolRestoreTest(BitcoinTestFramework):
     def set_test_params(self):
         self.setup_clean_chain = True
         self.num_nodes = 2
-        self.extra_args = [['-deprecatedrpc=accounts'], ['-deprecatedrpc=accounts', '-keypool=100', '-keypoolmin=20']]
+        self.extra_args = [[], ['-keypool=100', '-keypoolmin=20']]
```
Comment (MarcoFalke) [inline test/functional/wallet_keypool_topup.py]: What does keypoolmin do?

File: test/functional/wallet_listtransactions.py [hunk 3]
```diff
@@ -81,24 +80,23 @@ def run_test(self):
...
         self.nodes[0].importaddress(multisig["redeemScript"], "watchonly", False, True)
         txid = self.nodes[1].sendtoaddress(multisig["address"], 0.1)
         self.nodes[1].generate(1)
         self.sync_all()
-        assert(len(self.nodes[0].listtransactions("watchonly", 100, 0, False)) == 0)
-        assert_array_result(self.nodes[0].listtransactions("watchonly", 100, 0, True),
-                            {"category": "receive", "amount": Decimal("0.1")},
-                            {"txid": txid, "account": "watchonly"})
+        assert not [tx for tx in self.nodes[0].listtransactions("*", 100, 0, False) if "label" in tx and tx["label"] == "watchonly"]
+        txs = [tx for tx in self.nodes[0].listtransactions("*", 100, 0, True) if "label" in tx and tx['label'] == 'watchonly']
```
Comment (MarcoFalke) [inline test/functional/wallet_listtransactions.py]: nit: Could use named args for those `True` and `False`?

**Comment by MarcoFalke** [comment]
utACK 6b7a4ea7e9929d0ad65c7caf822f44ab6c01a475. Some style nits/questions.

File: test/functional/wallet_keypool_topup.py (hunk 2 above)
Comment (MarcoFalke) [reply test/functional/wallet_keypool_topup.py]: Looks like this was added in d34957e17e8c9740104533aaf4a896e93548c87e and it is fine to just remove.

**Comment by MeshCollider** [comment]
Concept ACK

File: test/functional/wallet_keypool_topup.py (hunk 2 above)
Comment (jnewbery) [reply test/functional/wallet_keypool_topup.py]: removed

File: test/functional/wallet_listtransactions.py (hunk 3 above)
Comment (jnewbery) [reply test/functional/wallet_listtransactions.py]: You can't mix positional and named args, so I've named them all.

**Comment by jnewbery** [comment]
Thanks for the reviews. I've addressed @MarcoFalke 's comments.

```
→ git diff 6b7a4ea7e9929d0ad65c7caf822f44ab6c01a475 5d536619abda745c298fd827a856df775f223241
diff --git a/test/functional/wallet_keypool_topup.py b/test/functional/wallet_keypool_topup.py
index 30a0c9a76..d657dc30c 100755
--- a/test/functional/wallet_keypool_topup.py
+++ b/test/functional/wallet_keypool_topup.py
@@ -25,7 +25,7 @@ class KeypoolRestoreTest(BitcoinTestFramework):
     def set_test_params(self):
         self.setup_clean_chain = True
         self.num_nodes = 2
-        self.extra_args = [[], ['-keypool=100', '-keypoolmin=20']]
+        self.extra_args = [[], ['-keypool=100']]
 
     def run_test(self):
         wallet_path = os.path.join(self.nodes[1].datadir, "regtest", "wallets", "wallet.dat")
diff --git a/test/functional/wallet_listtransactions.py b/test/functional/wallet_listtransactions.py
index 3de65b9b5..7cf2e456c 100755
--- a/test/functional/wallet_listtransactions.py
+++ b/test/functional/wallet_listtransactions.py
@@ -94,8 +94,8 @@ class ListTransactionsTest(BitcoinTestFramework):
         txid = self.nodes[1].sendtoaddress(multisig["address"], 0.1)
         self.nodes[1].generate(1)
         self.sync_all()
-        assert not [tx for tx in self.nodes[0].listtransactions("*", 100, 0, False) if "label" in tx and tx["label"] == "watchonly"]
-        txs = [tx for tx in self.nodes[0].listtransactions("*", 100, 0, True) if "label" in tx and tx['label'] == 'watchonly']
+        assert not [tx for tx in self.nodes[0].listtransactions(dummy="*", count=100, skip=0, include_watchonly=False) if "label" in tx and tx["label"] == "watchonly"]
+        txs = [tx for tx in self.nodes[0].listtransactions(dummy="*", count=100, skip=0, include_watchonly=True) if "label" in tx and tx['label'] == 'watchonly']
         assert_array_result(txs, {"category": "receive", "amount": Decimal("0.1")}, {"txid": txid})
 
         self.run_rbf_opt_in_test()
```
